src/psyclone/domain/lfric/lfric_stencils.py

```diff
@@ -0,0 +1,649 @@
+                ))
+
+    @property
+    def _unique_direction_vars(self):
+        '''
+        :returns: a list of all the unique direction argument names in this \
+                  invoke call.
+        :rtype: list of str
+        '''
+        names = []
+        for arg in self._unique_direction_args:
+            if arg.stencil.direction_arg.varname:
+                names.append(arg.stencil.direction_arg.varname)
+            else:
+                names.append(arg.name+"_direction")
+        return names
+
+    def _declare_unique_direction_vars(self, parent):
+        '''
+        Declare all unique direction arguments as integers with intent in
+        and add the declaration as a child of the parent argument
+        passed in.
+
+        :param parent: the node in the f2pygen AST to which to add the \
+                       declarations.
+        :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
+
+        '''
+        api_config = Config.get().api_conf("dynamo0.3")
+
+        if self._unique_direction_vars:
+            parent.add(DeclGen(parent, datatype="integer",
+                               kind=api_config.default_kind["integer"],
+                               entity_decls=self._unique_direction_vars,
+                               intent="in"))
+
+    @property
+    def unique_alg_vars(self):
+        '''
+        :returns: list of the names of the extent and direction arguments \
+                  supplied to the PSy routine from the Algorithm layer.
+        :rtype: list of str
+        '''
+        return self._unique_extent_vars + self._unique_direction_vars
+
+    def _invoke_declarations(self, parent):
+        '''
+        Declares all stencil maps, extent and direction arguments passed into
+        the PSy layer.
+
+        :param parent: node in the f2pygen AST to which to add declarations.
+        :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
+
+        '''
+        self._declare_unique_extent_vars(parent)
+        self._declare_unique_direction_vars(parent)
+        self._declare_maps_invoke(parent)
+
+    def _stub_declarations(self, parent):
+        '''
+        Declare all stencil-related quanitites for a Kernel stub.
+
+        :param parent: node in the f2pygen AST to which to add declarations.
+        :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
+
+        '''
+        self._declare_unique_extent_vars(parent)
+        self._declare_unique_direction_vars(parent)
+        self._declare_unique_max_branch_length_vars(parent)
+        self._declare_maps_stub(parent)
+
+    def initialise(self, parent):
+        '''
+        Adds in the code to initialise stencil dofmaps to the PSy layer.
+
+        :param parent: the node in the f2pygen AST to which to add the \
+                       initialisations.
+        :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
+
+        :raises GenerationError: if an unsupported stencil type is encountered.
+        '''
```

Comment:
Please insert a blank line above ‘’’ (please also do that through the module where missing).